### DISCUSSION TITLE
Why is the docker image still build with Python 3.7?

### DISCUSSION THREAD

**Discussion by empusas** [apache/airflow#31217] [General]
I wonder why the official docker images are still build on the old python 3.7 image. Not only it is listed that python 3.10 is supported, it comes with some nice features like match/case and is much more performant.

**Comment by potiuk** [comment]
As you likely know if you read our documentation, there are multiple images with Python 3.7 -> 3.10 (3.11 soon -we are waiting for dependencies to catch up). 

As you can read in our policies https://airflow.apache.org/docs/docker-stack/index.html default image is the latest supported version (this is our agreed policy). And you are absolutely free to use specific version images if you want.

The policy is to handle those who prefer stability over fanciness. If you prefer fanciness and new shiny things - fee free to use the `-python3.10` version. Absolutely no-one can stop you from doing it.

And if you have a reasoning why you think our policy could be changed - also feel absolutely free to discuss, get consensus and voting on changing the policy. Subscribe to devlist, write a "[PROPOSAL]" email when you will propose a different policy and justify why you think different policy will be better.  

You are welcome to start such discussion - due to the fact that this is agreed and voted policies and that it is a public policy that our users rely on, it has to be done on the devlist and this is the only place it should be discussed.

**Reply by empusas** [reply]
Hi, thanks a lot. Just downloaded the python 3.10 image.

**Comment by potiuk** [comment]
BTW. If you actually read our policy about supporting Python version, the first MINOR release after 30 June (when Python 3.7 EOL is) will be based on Python 3.8.

If you need more reasoning and why's for our versioning policies of various components - you will find them nicely described (together with reasoning) in https://github.com/apache/airflow#version-life-cycle
